-1

**Failed Tests**: AddOn 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-69bdc3/28211/summary.html
**COMMIT**: c78fc70c616d6b11e31a1918a5b9b067e0c227c9
**CMSSW**: CMSSW_12_6_X_2022-10-12-1100/el8_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/39699/28211/install.sh` to create a dev area with all the needed externals and cmssw changes.


## AddOn Tests

- [read312RV](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-69bdc3/28211/addOnTests/read312RV)
```
----- Begin Fatal Exception 12-Oct-2022 23:27:27 CEST-----------------------
An exception of category 'FileOpenError' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing input source of type PoolSource
   [2] Calling RootInputFileSequence::initTheFile()
   [3] Calling StorageFactory::open()
   [4] Calling XrdFile::open()
Exception Message:
Failed to open the file 'root://xrootd-cms.infn.it//store/data/Run2010A/MinimumBias/RAW/v1/000/137/289/B419A444-BD73-DF11-A22C-0030487CD812.root'
   Additional Info:
      [a] Calling RootInputFileSequence::initTheFile(): fail to open the file with name root://eoscms.cern.ch//eos/cms/store/data/Run2010A/MinimumBias/RAW/v1/000/137/289/B419A444-BD73-DF11-A22C-0030487CD812.root
      [b] Calling RootInputFileSequence::initTheFile(): fail to open the file with name root://eoscms.cern.ch//eos/cms/store/user/cmsbuild/store/data/Run2010A/MinimumBias/RAW/v1/000/137/289/B419A444-BD73-DF11-A22C-0030487CD812.root
      [c] Input file root://xrootd-cms.infn.it//store/data/Run2010A/MinimumBias/RAW/v1/000/137/289/B419A444-BD73-DF11-A22C-0030487CD812.root could not be opened.
      [d] XrdCl::File::Open(name='root://xrootd-cms.infn.it//store/data/Run2010A/MinimumBias/RAW/v1/000/137/289/B419A444-BD73-DF11-A22C-0030487CD812.root', flags=0x10, permissions=0660) => error '[ERROR] Operation expired' (errno=0, code=206). No additional data servers were found.
      [e] Last URL tried: root://cms-xrd-global.cern.ch:1094//store/data/Run2010A/MinimumBias/RAW/v1/000/137/289/B419A444-BD73-DF11-A22C-0030487CD812.root?tried=+1213xrootd-cms-redir-int.cr.cnaf.infn.it,+1213llrxrd-redir.in2p3.fr,,eoscms-ns-ip563.cern.ch,eoscms-ns-ip563.cern.ch,xrootd-redic.pi.infn.it&xrdcl.requuid=fff9d36d-5fe3-4705-b774-d912e9f53b8d
      [f] Problematic data server: cms-xrd-global.cern.ch:1094
      [g] Disabled source: cms-xrd-global.cern.ch:1094
----- End Fatal Exception -------------------------------------------------
```
- [pat1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-69bdc3/28211/addOnTests/pat1)
```
----- Begin Fatal Exception 12-Oct-2022 23:22:55 CEST-----------------------
An exception of category 'FileOpenError' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing input source of type PoolSource
   [2] Calling RootInputFileSequence::initTheFile()
   [3] Calling StorageFactory::open()
   [4] Calling XrdFile::open()
Exception Message:
Failed to open the file 'root://xrootd-cms.infn.it//store/relval/CMSSW_9_2_2/RelValProdTTbar_13/AODSIM/91X_mcRun2_asymptotic_v3-v1/10000/EEB99F74-DA4D-E711-A41C-0025905A48F2.root'
   Additional Info:
      [a] Calling RootInputFileSequence::initTheFile(): fail to open the file with name root://eoscms.cern.ch//eos/cms/store/relval/CMSSW_9_2_2/RelValProdTTbar_13/AODSIM/91X_mcRun2_asymptotic_v3-v1/10000/EEB99F74-DA4D-E711-A41C-0025905A48F2.root
      [b] Calling RootInputFileSequence::initTheFile(): fail to open the file with name root://eoscms.cern.ch//eos/cms/store/user/cmsbuild/store/relval/CMSSW_9_2_2/RelValProdTTbar_13/AODSIM/91X_mcRun2_asymptotic_v3-v1/10000/EEB99F74-DA4D-E711-A41C-0025905A48F2.root
      [c] Input file root://xrootd-cms.infn.it//store/relval/CMSSW_9_2_2/RelValProdTTbar_13/AODSIM/91X_mcRun2_asymptotic_v3-v1/10000/EEB99F74-DA4D-E711-A41C-0025905A48F2.root could not be opened.
      [d] XrdCl::File::Open(name='root://xrootd-cms.infn.it//store/relval/CMSSW_9_2_2/RelValProdTTbar_13/AODSIM/91X_mcRun2_asymptotic_v3-v1/10000/EEB99F74-DA4D-E711-A41C-0025905A48F2.root', flags=0x10, permissions=0660) => error '[ERROR] Server responded with an error: [3011] No servers are available to read the file.
' (errno=3011, code=400). No additional data servers were found.
      [e] Last URL tried: root://cms-xrd-global.cern.ch:1094//store/relval/CMSSW_9_2_2/RelValProdTTbar_13/AODSIM/91X_mcRun2_asymptotic_v3-v1/10000/EEB99F74-DA4D-E711-A41C-0025905A48F2.root?tried=+1213xrootd-cms-redir-int.cr.cnaf.infn.it,&xrdcl.requuid=c8b0bef3-4964-4b7f-8381-bfb88f0d53cf
      [f] Problematic data server: cms-xrd-global.cern.ch:1094
      [g] Disabled source: cms-xrd-global.cern.ch:1094
----- End Fatal Exception -------------------------------------------------
```
- [hlt_data_Fake](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-69bdc3/28211/addOnTests/hlt_data_Fake)
```
----- Begin Fatal Exception 12-Oct-2022 23:28:16 CEST-----------------------
An exception of category 'FileOpenError' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing input source of type PoolSource
   [2] Calling RootInputFileSequence::initTheFile()
   [3] Calling StorageFactory::open()
   [4] Calling XrdFile::open()
Exception Message:
Failed to open the file 'root://xrootd-cms.infn.it//store/data/Run2012A/MuEG/RAW/v1/000/191/718/14932935-E289-E111-830C-5404A6388697.root'
   Additional Info:
      [a] Calling RootInputFileSequence::initTheFile(): fail to open the file with name root://eoscms.cern.ch//eos/cms/store/data/Run2012A/MuEG/RAW/v1/000/191/718/14932935-E289-E111-830C-5404A6388697.root
      [b] Calling RootInputFileSequence::initTheFile(): fail to open the file with name root://eoscms.cern.ch//eos/cms/store/user/cmsbuild/store/data/Run2012A/MuEG/RAW/v1/000/191/718/14932935-E289-E111-830C-5404A6388697.root
      [c] Input file root://xrootd-cms.infn.it//store/data/Run2012A/MuEG/RAW/v1/000/191/718/14932935-E289-E111-830C-5404A6388697.root could not be opened.
      [d] XrdCl::File::Open(name='root://xrootd-cms.infn.it//store/data/Run2012A/MuEG/RAW/v1/000/191/718/14932935-E289-E111-830C-5404A6388697.root', flags=0x10, permissions=0660) => error '[ERROR] Operation expired' (errno=0, code=206). No additional data servers were found.
      [e] Last URL tried: root://cms-xrd-global.cern.ch:1094//store/data/Run2012A/MuEG/RAW/v1/000/191/718/14932935-E289-E111-830C-5404A6388697.root?tried=+1213xrootd-redic.pi.infn.it,+1213llrxrd-redir.in2p3.fr,,eoscms-ns-ip563.cern.ch,eoscms-ns-ip563.cern.ch,xrootd-cms-redir-int.cr.cnaf.infn.it&xrdcl.requuid=d532f21e-41d0-4e5c-bc4e-df00a289ddf1
      [f] Problematic data server: cms-xrd-global.cern.ch:1094
      [g] Disabled source: cms-xrd-global.cern.ch:1094
----- End Fatal Exception -------------------------------------------------
```
<details>
<summary>Expand to see more addon errors ...</summary>

- [hlt_mc_Fake2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-69bdc3/28211/addOnTests/hlt_mc_Fake2)
- [hlt_data_Fake](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-69bdc3/28211/addOnTests/hlt_data_Fake)
- [hlt_data_Fake1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-69bdc3/28211/addOnTests/hlt_data_Fake1)
- [hlt_mc_Fake2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-69bdc3/28211/addOnTests/hlt_mc_Fake2)
- [hlt_data_Fake1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-69bdc3/28211/addOnTests/hlt_data_Fake1)
- [hlt_data_Fake2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-69bdc3/28211/addOnTests/hlt_data_Fake2)
- [hlt_mc_Fake2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-69bdc3/28211/addOnTests/hlt_mc_Fake2)
- [hlt_data_Fake2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-69bdc3/28211/addOnTests/hlt_data_Fake2)
- [hlt_data_GRun](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-69bdc3/28211/addOnTests/hlt_data_GRun)
- [hlt_mc_GRun](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-69bdc3/28211/addOnTests/hlt_mc_GRun)
- [hlt_data_GRun](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-69bdc3/28211/addOnTests/hlt_data_GRun)
- [hlt_data_HIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-69bdc3/28211/addOnTests/hlt_data_HIon)
- [hlt_mc_HIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-69bdc3/28211/addOnTests/hlt_mc_HIon)
- [hlt_data_HIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-69bdc3/28211/addOnTests/hlt_data_HIon)
- [hlt_data_PIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-69bdc3/28211/addOnTests/hlt_data_PIon)
- [hlt_mc_PIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-69bdc3/28211/addOnTests/hlt_mc_PIon)
- [hlt_data_PIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-69bdc3/28211/addOnTests/hlt_data_PIon)
- [hlt_data_PRef](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-69bdc3/28211/addOnTests/hlt_data_PRef)
- [hlt_mc_PRef](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-69bdc3/28211/addOnTests/hlt_mc_PRef)
- [hlt_data_PRef](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-69bdc3/28211/addOnTests/hlt_data_PRef)
</details>



## Comparison Summary



@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-69bdc3/20834.0_TTbar_14TeV+2026D88+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-69bdc3/20834.75_TTbar_14TeV+2026D88_HLT75e33+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HLT75e33+HARVESTGlobal
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-69bdc3/20896.0_CloseByPGun_CE_E_Front_120um+2026D88+CE_E_Front_120um_GenSimHLBeamSpotHGCALCloseBy+DigiTrigger+RecoGlobal+HARVESTGlobal
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-69bdc3/20900.0_CloseByPGun_CE_H_Coarse_Scint+2026D88+CE_H_Coarse_Scint_GenSimHLBeamSpotHGCALCloseBy+DigiTrigger+RecoGlobal+HARVESTGlobal
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-69bdc3/21034.999_TTbar_14TeV+2026D88PU_PMXS1S2PR+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+PREMIX_PremixHLBeamSpot14PU+DigiTriggerPU+RecoGlobalPU+HARVESTGlobalPU
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-69bdc3/23234.0_TTbar_14TeV+2026D94+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal

**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 6 differences found in the comparisons
 - DQMHistoTests: Total files compared: 49
 - DQMHistoTests: Total histograms compared: 3392309
 - DQMHistoTests: Total failures: 9
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3392278
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 48 files compared)
 - Checked 204 log files, 49 edm output root files, 49 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_6_X_2022-10-12-1100+69bdc3/53357/triggerResults): no differences found

--- a/SimPPS/RPDigiProducer/plugins/RPDisplacementGenerator.cc
+++ b/SimPPS/RPDigiProducer/plugins/RPDisplacementGenerator.cc
@@ -17,15 +17,7 @@ RPDisplacementGenerator::RPDisplacementGenerator(bool iIsOn,
                                                  const CTPPSRPAlignmentCorrectionsData *alignments,
                                                  const CTPPSGeometry &geom)
     : detId_(_detId) {
-  isOn_ = iIsOn;  //ps.getParameter<bool>("RPDisplacementOn");
-
-  // read the alignment correction
-  /*
-  ESHandle<CTPPSRPAlignmentCorrectionsData> alignments;
-  if (auto rec = iSetup.tryToGet<VeryForwardMisalignedGeometryRecord>()) {
-    iSetup.get<VeryForwardMisalignedGeometryRecord>().get(alignments);
-  }
-  */
+  isOn_ = iIsOn;
 
   unsigned int decId = rawToDecId(detId_);
 
@@ -40,8 +32,6 @@ RPDisplacementGenerator::RPDisplacementGenerator(bool iIsOn,
     isOn_ = false;
 
   // transform shift and rotation to the local coordinate frame
-  //ESHandle<CTPPSGeometry> geom;
-  //iSetup.get<VeryForwardRealGeometryRecord>().get(geom);
   const DetGeomDesc *g = geom.sensor(detId_);
   const RotationMatrix &R_l = g->rotation();
   rotation_ = R_l.Inverse() * R_m.Inverse() * R_l;

